driver-core/src/main/java/com/datastax/driver/core/TupleType.java

```diff
@@ -124,6 +136,32 @@ public boolean equals(Object o) {
         return name == d.name && types.equals(d.types);
     }
 
+    /**
+     * Return {@code true} if this tuple type contains the given tuple type,
+     * and {@code false} otherwise.
+     * <p>
+     * A tuple type is said to contain another one
+     * if the latter has fewer components than the former,
+     * but all of them are of the same type.
+     * E.g. the type {@code tuple<int, text>}
+     * is contained by the type {@code tuple<int, text, double>}.
+     * <p>
+     * A contained type can be seen as a "partial" view
+     * of a containing type, where the missing components
+     * are supposed to be {@code null}.
+     *
+     * @param other the tuple type to compare against the current one
+     * @return {@code true} if this tuple type contains the given tuple type,
+     * and {@code false} otherwise.
+     */
+    public boolean contains(TupleType other) {
```

Comment:
I understand how this method is useful for `TupleCodec`, but I'm not sure if it should be public.
Is it because users writing a generic tuple codec would need it too? (sorry for my lack of perspective here)